### PR TITLE
Extend conventions and guidelines

### PR DESCRIPTION
**Pull request by Cr3ature**
- extend with codespaces and docker
- reform github pages 
- reform navigations 
- extend pages

### CODE REVIEW

File: _site/code/index.html [hunk 1]
```diff
@@ -0,0 +1,316 @@
+<!doctype html>
```
Comment (huysentruitw) [inline _site/code/index.html:1]: I would ignore the _site folder and exclude it from the repo, is that possible?

Comment (Cr3ature) [reply _site/code/index.html:1]: should be possible in the ignore folder. This is build from within the codespaces. Added in the .gitignore

File: _pages/code/codeconventions/best-practices.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+### Modelstate
+
+In the controller, we can check if the `modelstate` is valid. `ModelState.IsValid` determines wether the submitted values statisfy all `DataAnnotation` validation attributes applied to the model properties.
+
+```csharp
+[HttpPost]
+public ActionResult CreateUser(CreateUserViewModel createUserForm)
+{
+    if(!ModelState.IsValid)
```
Comment (ThomasVer) [inline _pages/code/codeconventions/best-practices.md:186]:
```suggestion
    if (!ModelState.IsValid)
```

File: _pages/code/codeconventions/best-practices.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    {
+        //Code to handle invalid form
+    }
+
+    //Code to create user
```
Comment (ThomasVer) [inline _pages/code/codeconventions/best-practices.md:191]:
```suggestion
    // Code to create user
```

File: _pages/code/codeconventions/pattern-matching.md [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+- `constant` pattern: tests if an expression equals a specified constant. This can be a value, a declared const variable or an enumeration constant. Also checking for null is perfomed via constant pattern.
+
+```csharp
+const int threeDiceHighRoll = 18;
+
+if(diceRoll is threeDiceHighRoll ){
+  Console.WriteLine("To the max!");
+}
+
+if(diceRoll is 3){
```
Comment (ThomasVer) [inline _pages/code/codeconventions/pattern-matching.md:42]:
```suggestion
if (diceRoll is 3)
{
```

File: _pages/tooling/editorConfig/coding-conventions.md [hunk 5]
```diff
@@ -1,15 +1,18 @@
...
 sidebar:
-  nav: "editorconfig"
+  nav: "tooling"
 
 toc: true
 toc_sticky: true
 ---
 
+## EditorConfig file
+nog wa nieuws
```
Comment (ThomasVer) [inline _pages/tooling/editorConfig/coding-conventions.md:14]: Add actual content

File: _pages/code/codeguidelines/coding-style.md [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+Avoid using '+' to concatenate text into a new string.
+
+```csharp
+string name = "John";
+string greetings = "Hello " + name + "!";
+
+//Good: string.Format()
+string greetingFormat = string.Format("Hello {0}!", name);
+
+//Better: string interpolation (${})
```
Comment (ThomasVer) [inline _pages/code/codeguidelines/coding-style.md:337]:
```suggestion
// Better: string interpolation (${})
```

File: _pages/code/codeconventions/pattern-matching.md [hunk 7]
```diff
@@ -0,0 +1,347 @@
...
+      Console.Writeline("obj is an integer with value {number}");
+  default:
+      throw new ArgumentException("Invalid data");
+}
+```
+
+`When` pattern allows you to have additional conditions on the switch case.
+
+```csharp
+switch(number)
```
Comment (ThomasVer) [inline _pages/code/codeconventions/pattern-matching.md:82]:
```suggestion
switch (number)
```

File: _pages/code/codeconventions/pattern-matching.md [hunk 8]
```diff
@@ -0,0 +1,347 @@
...
+      Console.WriteLine("More than 0 but less than or equal to 10");
+      break;
+  default:
+      Console.WriteLine("More than 10");
+      break;
+}
+```
+
+```csharp
+switch(shape)
```
Comment (ThomasVer) [inline _pages/code/codeconventions/pattern-matching.md:97]:
```suggestion
switch (shape)
```

**Review by ThomasVer — CHANGES_REQUESTED** [review]

File: CONDUCT.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+* Spamming, trolling, intentionally disrupting conversations, or irrelevant solicitation or advertisement
+* Making demeaning or discriminatory comments
+* Making negative assumptions about someone's background, abilities, or intentions
+* Harassing or stalking individuals (online or in person)
+* Giving someone unwelcome sexual attention or making unwelcome physical contact (in the case of an IRL event)
+* Sharing sexual images or using sexually explicit language
+
+In general: treat others how you would like to be treated, were you in their place. Don't be a jerk. _Do_ ask questions. _Do_ keep conflicts productively focused on technical issues. _Do_ think before you speak; remember that what is perceived as a funny witticism in your group of friends might be hurtful or reinforce hurtful stereotypes in the context of our diverse online community. _Do_ remember that we are all people, not robots, and all equally deserving of sensitivity and respect. (If and when robots join our community, let's treat them with respect too!)
+
+
```
[the remark this answers is not in the document]
Comment (Cr3ature) [reply CONDUCT.md:20]: resolved in new commit

File: _pages/code/codeconventions/addition-to-conventions.md [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+
+```csharp
+// Explicit cast
+object obj = "Noest";
+var word = (string)obj;
+
+SomeClass someObject = (SomeClass)obj;
+
+
+//As operator
```
[the remark this answers is not in the document]
Comment (Cr3ature) [reply _pages/code/codeconventions/addition-to-conventions.md:108]: Resolved in new commit

File: _pages/code/codeconventions/best-practices.md (hunk 2 above)
Comment (Cr3ature) [reply _pages/code/codeconventions/best-practices.md:186]: Resolved in next commit

File: _pages/code/codeconventions/best-practices.md [hunk 11]
```diff
@@ -0,0 +1,193 @@
...
+
+In the controller, we can check if the `modelstate` is valid. `ModelState.IsValid` determines wether the submitted values statisfy all `DataAnnotation` validation attributes applied to the model properties.
+
+```csharp
+[HttpPost]
+public ActionResult CreateUser(CreateUserViewModel createUserForm)
+{
+    if(!ModelState.IsValid)
+    {
+        //Code to handle invalid form
```
[the remark this answers is not in the document]
Comment (Cr3ature) [reply _pages/code/codeconventions/best-practices.md:188]: Resolved in next commit

File: _pages/code/codeconventions/best-practices.md (hunk 3 above)
Comment (Cr3ature) [reply _pages/code/codeconventions/best-practices.md:191]: Resolved in next commit

**Review by ThomasVer — DISMISSED** [review]

**Pull request merged**